hack/nix/overlay-kind.nix

```diff
@@ -0,0 +1,18 @@
+final: prev: {
+  # Overlay kind to version v0.15.0
```

Comment:
Hmm.. this looks like an interesting file, but I'm not totally sure what it does or how it works 👀 can you add some comments, and possibly links to any relevant articles to help understand some of this?

Reply:
Sure thing!

To share here also: this file contains a nix ["overlay"](https://nixos.wiki/wiki/Overlays). Overlays are the canonical method in nix to overwrite definitions. In this case we are overriding kind from version `v0.14.0` to `v1.15.0`. This is a good way for using newer versions than that in nixpkgs or adding patches etc.

As I wrote this message I see that kind has now been updated to v0.15.0 upstream so this overlay can be removed.